awx/ui/src/screens/Team/TeamAdd/TeamAdd.js

```diff
@@ -1,60 +1,50 @@
-import React from 'react';
-import { withRouter } from 'react-router-dom';
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
 import { PageSection, Card } from '@patternfly/react-core';
 
 import { TeamsAPI } from 'api';
 import { Config } from 'contexts/Config';
 import { CardBody } from 'components/Card';
 import TeamForm from '../shared/TeamForm';
 
-class TeamAdd extends React.Component {
-  constructor(props) {
-    super(props);
-    this.handleSubmit = this.handleSubmit.bind(this);
-    this.handleCancel = this.handleCancel.bind(this);
-    this.state = { error: null };
-  }
+function TeamAdd() {
+  const [submitError, setSubmitError] = useState(null);
+  const history = useHistory();
 
-  async handleSubmit(values) {
-    const { history } = this.props;
+  const handleSubmit = async (values) => {
     try {
       const valuesToSend = { ...values };
```

Comment:
I would probably prefer destructuring here to avoid data mutation on line `17`.

```javascript
const { name, description, organization: { id } } = values;
const valuesToSend = { name, description, organization: id };
```

Reply:
@kialam updated per comments.